torch_xla/core/dynamo_bridge.py

```diff
@@ -111,6 +111,65 @@ def get_fallback_ops():
   return fallback_ops
 
 
+# Checks that all input args that are tensors are on the same device.
+def _get_input_arg_device(input_args: tuple) -> torch.device:
+  device = None
+  for arg in input_args:
+    if not isinstance(arg, torch.Tensor):
+      continue
+
+    if device is None:
+      device = arg.device
+    else:
+      assert arg.device == device, "Not all args are on the same device."
```

Comment:
@ysiraichi I remembered you added a pass to dynamo bridge to handle where some of the inputs are on cpu(by graph partiton). Will the change here break that?

Reply:
I think this is fine. If I remember correctly `ConstructorMoverPass` only modified tensor constructor nodes (not arguments).